compiler/jsgen.nim

```diff
@@ -179,9 +179,24 @@ proc mapType(p: PProc; typ: PType): TJSTypeKind =
   else: result = mapType(typ)
 
 proc mangleName(s: PSym; target: TTarget): Rope =
+  proc validJsName(name: string): bool =
+    result = true
+    const reservedWords = ["break", "case", "catch", "class", "const", "continue",
```

Comment:
Even though the list looks correct, it may be shrinked by excluding nim words, which are already guaranteed not to be present on the parsing phase. This will somewhat improve the lookup speed, as its currently O(n).